src/scaffold.ts

```diff
@@ -92,21 +135,15 @@ export const scaffoldProjectRequest = async (item?: KafkaCluster | KafkaTopic) =
    */
   let optionValues: { [key: string]: string | boolean } = {};
   let options = templateSpec.options || {};
-  async function getInitialOptionValue(option: string, properties: any): Promise<string | boolean> {
-    if ((option === "topic" || option === "cc_topic") && item instanceof KafkaTopic)
-      return item.name;
-    else if (option === "bootstrap_server" || option === "cc_bootstrap_server") {
-      if (item instanceof KafkaCluster) return item.bootstrapServers;
-      else if (item instanceof KafkaTopic) {
-        const cluster = await getResourceManager().getClusterForTopic(item);
-        return cluster?.bootstrapServers ?? "";
-      }
+
+  for (const option of Object.keys(options)) {
+    if (templateRequestOptions && templateRequestOptions[option] !== undefined) {
+      optionValues[option] = templateRequestOptions[option] as string | boolean; // Explicitly cast to string | boolean
+    } else {
+      optionValues[option] = ""; // Provide a default value for undefined options
```

Comment:
@Cerchie this should properly handle conversion to booleans while falling back to `initial_value` if available for the template:
```suggestion
  for (const [option, properties] of Object.entries(options)) {
    if (templateRequestOptions && templateRequestOptions[option] !== undefined) {
      let value: string | boolean = "";
      if (["true", "false"].includes(value)) {
        // convert to boolean
        value = templateRequestOptions[option] === "true";
      } else {
        // let everything else through as a string
        value = templateRequestOptions[option];
      }
      optionValues[option] = value;
    } else {
      optionValues[option] = properties.initial_value ?? "";
    }
  }
```